src/controllers/SafeNavigationBaseController.ts

```diff
@@ -0,0 +1,97 @@
+import { AnySchema } from '@hapi/joi';
+import { SessionStore } from 'ch-node-session-handler';
+import { Cookie } from 'ch-node-session-handler/lib/session/model/Cookie';
+import { Request } from 'express';
+import { inject } from 'inversify';
+import { provide } from 'inversify-binding-decorators';
+
+import { BaseController, FormSanitizeFunction } from 'app/controllers/BaseController';
+import {
+    FormSubmissionProcessor,
+    FormSubmissionProcessorConstructor
+} from 'app/controllers/processors/FormSubmissionProcessor';
+import { ApplicationData, APPEALS_KEY } from 'app/models/Appeal';
+import { getEnvOrDefault } from 'app/utils/EnvironmentUtils';
+import { PENALTY_DETAILS_PAGE_URI } from 'app/utils/Paths';
+import { Navigation } from 'app/utils/navigation/navigation';
+
+type RequestWithNavigation = Request & { navigation: Navigation }
+
+@provide(InternalProcessor)
+class InternalProcessor implements FormSubmissionProcessor {
```

Comment:
I think name `Processor` would work better. Word 'internal' only adds to the confusion.